packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -491,6 +496,14 @@ class AndroidCameraCameraX extends CameraPlatform {
   /// Sets the flash mode for the selected camera.
   @override
   Future<void> setFlashMode(int cameraId, FlashMode mode) async {
+    CameraControl? cameraControl;
+    // Turn off torch mode if it is enabled and not being redundantly set.
+    if (mode != FlashMode.torch && torchEnabled) {
```

Comment:
As currently written, we would have different behavior for 
```
...
setFlashMode(camera_id, FlashMode.auto)
setFlashMode(camera_id, FlashMode.torch)
takePicture(camera_id)
// results in taking a picture with both underlying camerax auto flash mode, and torch mode on
```
and
```
...
setFlashMode(camera_id, FlashMode.torch)
setFlashMode(camera_id, FlashMode.auto)
takePicture(camera_id)
// results in taking a picture with torch mode off and auto flash mode on
```
which could be confusing (I don't think this would be confusing if it were always just the case that the mode set last is the enabled one, but in this case the result is that you either get both set, or only one, depending on order).

I'm not sure if the best way to fix it is just to make the documentation very clear, or to do a similar thing where we unset the flash modes when setting the torch mode (like we unset the torch mode when setting the flash). What do you think?

All of the awkwardness seems to come from the fact that camerax behaves differently here, where flash and torch are independent configurations that can both be enabled or disabled at the same time. But it seems like we can't do that without adding a `FlashMode.torchOn` and `FlashMode.torchOff` to the type.

Reply:
Also most camerax examples seem to guard the setting of the flashmode with a [`hasFlashUnit`](https://developer.android.com/reference/androidx/camera/core/CameraInfo#hasFlashUnit()) call on the corresponding cameraInfo. 

It looks like both setting the torch and flash mode will just take no effect if there is no flash unit, so it probably isn't crucial to include, but it would be good to note that behavior in the documentation here

Reply:
To your first point about the behavior differences, I don't think that is true because when non-torch modes are set, it is saved and used when taking a picture. Setting the torch sets that saved mode to null, so setting auto then torch will clear auto as the saved setting.

As to the `hasFlashUnit` gating that seems to happen, I'l document that behavior and leave a todo to come back and use it for better error handling.

Reply:
Oh wait I see what you're saying on the behavior difference in terms of CameraX. Yeah I'll update this PR to clear the flash mode when the mode isn't changed because that is undesired behavior.

Reply:
Yeah it's actually not true exactly as written, but if I change it to 

```
...
setFlashMode(camera_id, FlashMode.auto)
takePicture(camera_id)
setFlashMode(camera_id, FlashMode.torch)
takePicture(camera_id)
// results in taking a picture with both underlying camerax auto flash mode, and torch mode on
```
and
```
...
setFlashMode(camera_id, FlashMode.torch)
takePicture(camera_id)
setFlashMode(camera_id, FlashMode.auto)
takePicture(camera_id)
// results in taking a picture with torch mode off and auto flash mode on
```

then it should be true, right? Because we [don't update the flash mode](https://github.com/flutter/packages/blob/bb0e9d76b998fb3b858b9304fcd12becd098c051/packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart#L489) used by the image capture after we set the `_currentFlashMode` to null.

Reply:
Right yeah I'm going to make sure we update the flash mode to be off if the torch is on and the saved mode is null.